src/js/tls.js

```diff
@@ -3,85 +3,114 @@
 require('crypto');
 
 var net = require('net');
+var tcp = require('tcp');
 var util = require('util');
-var EventEmitter = require('events').EventEmitter;
 var TlsWrap = native.TlsWrap;
 var TLS_CHUNK_MAX_SIZE = require('constants').TLS_CHUNK_MAX_SIZE;
 
-function TLSSocket(socket, opts) {
-  if (!(this instanceof TLSSocket))
-    return new TLSSocket(socket, opts);
-  EventEmitter.call(this);
+function Delegator(proto, target) {
+  this.proto = proto;
+  this.target = target;
+}
 
-  var tlsOptions = Object.assign({
-    servername: socket.host || socket.hostname,
-    // rejectUnauthorized: false,
-  }, opts);
+Delegator.prototype.method = function method(name) {
+  var proto = this.proto;
+  var target = this.target;
 
-  // TODO: currently certification is disabled
-  tlsOptions.rejectUnauthorized = false;
+  proto[name] = function delegation() {
+    return this[target][name].apply(this[target], arguments);
+  };
 
-  // handle the [ca1,ca2,...]
-  if (Array.isArray(tlsOptions.ca)) {
-    tlsOptions.ca = tlsOptions.ca.join('\n');
-  }
+  return this;
+};
 
-  this.servername = tlsOptions.servername;
-  this.authorized = false;
-  this.authorizationError = null;
-  this._socket = new net.Socket(socket);
-  this._writev = [];
-  this._ended = false;
+function TLSHandle(options) {
+  var self = this;
+  this.tcpHandle = new tcp();
+  this.tlsWrap = new TlsWrap(options);
+  this.tlsWrap.onwrite = function onwrite(chunk) {
+    self.tcpHandle.write(chunk, function afterWrite(status) {
+    });
+  };
+  this.tlsWrap.onread = function onread(nread, buffer) {
+    if (self.onread) {
+      return self.onread(self.tlsWrap.jsref, nread, self._EOF === true, buffer);
+    }
+  };
+  this.tcpHandle.onclose = this.tlsWrap.onclose = function onclose() {
+    if (self.onclose) {
+      self.onclose();
+    }
+  };
+
+  Object.defineProperty(this, 'owner', {
+    get: function get() {
```

Comment:
use `getOwner`.

Reply:
We have an eslint rule requires method name to be aligned with property name, i.e. we have to use `get` or alter our eslint rule.